homeassistant/components/evohome/__init__.py

```diff
@@ -160,136 +232,104 @@ def _handle_exception(err) -> bool:
-            await self._load_auth_tokens()
-        )
-
-        # evohomeasync2 uses naive/local datetimes
-        if access_token_expires is not None:
-            access_token_expires = _dt_to_local_naive(access_token_expires)
-
-        client = self.client = evohomeasync2.EvohomeClient(
-            self.params[CONF_USERNAME],
-            self.params[CONF_PASSWORD],
-            refresh_token=refresh_token,
-            access_token=access_token,
-            access_token_expires=access_token_expires,
-            session=async_get_clientsession(self.hass),
-        )
-
-        try:
-            await client.login()
-        except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
-            if not _handle_exception(err):
-                return False
-
-        finally:
-            self.params[CONF_PASSWORD] = "REDACTED"
-
-        self.hass.add_job(self._save_auth_tokens())
-
-        loc_idx = self.params[CONF_LOCATION_IDX]
-        try:
-            self.config = client.installation_info[loc_idx][GWS][0][TCS][0]
-
-        except IndexError:
-            _LOGGER.error(
-                "Config error: '%s' = %s, but its valid range is 0-%s. "
-                "Unable to continue. "
-                "Fix any configuration errors and restart HA.",
-                CONF_LOCATION_IDX,
-                loc_idx,
-                len(client.installation_info) - 1,
-            )
-            return False
 
+        loc_idx = params[CONF_LOCATION_IDX]
+        self.config = client.installation_info[loc_idx][GWS][0][TCS][0]
         self.tcs = (
             client.locations[loc_idx]  # pylint: disable=protected-access
             ._gateways[0]
             ._control_systems[0]
         )
+        self.temps = None
 
-        _LOGGER.debug("Config = %s", self.config)
-        if _LOGGER.isEnabledFor(logging.DEBUG):  # don't do an I/O unless required
-            await self.update()  # includes: _LOGGER.debug("Status = %s"...
-
-        return True
-
-    async def _load_auth_tokens(
-        self
-    ) -> Tuple[Optional[str], Optional[str], Optional[datetime]]:
-        store = self.hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
-        app_storage = self._app_storage = await store.async_load()
-
-        if app_storage is None:
-            app_storage = self._app_storage = {}
-
-        if app_storage.get(CONF_USERNAME) == self.params[CONF_USERNAME]:
-            refresh_token = app_storage.get(CONF_REFRESH_TOKEN)
-            access_token = app_storage.get(CONF_ACCESS_TOKEN)
-            at_expires_str = app_storage.get(CONF_ACCESS_TOKEN_EXPIRES)
-            if at_expires_str:
-                at_expires_dt = dt_util.parse_datetime(at_expires_str)
-            else:
-                at_expires_dt = None
-
-            return (refresh_token, access_token, at_expires_dt)
-
-        return (None, None, None)  # account switched: so tokens wont be valid
+        self.hass.add_job(self._save_auth_tokens())
```

Comment:
We don't want side effects in init method. Please move it out of the init method.

Reply:
We're in async context here so we have to use the async api. But we cant await inside init method. Another reason to not have side effects in init method.

Reply:
Done.